lib/cc/engine/analyzers/reporter.rb

```diff
@@ -0,0 +1,67 @@
+require 'cc/engine/analyzers/violation'
+
+module CC
+  module Engine
+    module Analyzers
+      class Reporter
+        def initialize(directory, language_strategy, io)
+          @directory = directory
+          @language_strategy = language_strategy
+          @io = io
+          @flay = Flay.new(flay_options)
```

Comment:
Yeah, it's a bit coupled but I think it makes sense.

I'll extract it into a private method which should make it a bit better.